util/signing/signer/src/lib.rs

```diff
@@ -0,0 +1,124 @@
+pub mod cryptography;
+
+/// A collection of bytes.
+#[derive(Debug, Clone)]
+pub struct Bytes(pub Vec<u8>);
+
+/// A signature.
+#[derive(Debug, Clone)]
+pub struct Signature(pub Bytes);
+
+/// A public key.
+#[derive(Debug, Clone)]
+pub struct PublicKey(pub Bytes);
```

Comment:
Yes, agree, it can be a vec<u8> for simplicity or an [u8, N] with the size as const generic. In this case, we should define type for the mostly used size.

Reply:
This sort of thing I believe is going to require associated types.